### PR TITLE
feat(oracle): replace signer with consensuskey and use ed25519 to sign the message

### PR DESCRIPTION
**Pull request by leonz789**
## Description

Use consensuskey as signing key to sign create-price message, and add corresponding support for ed25519.
This PR need to work with PR #63 . It should be merged after PR #63 

## Changes
Notable changes:
- add file app/ante/cosmos/sigverify.go to do customize modification
  - skip auth's account check, number, sequence
  - verify signature with algorithm ed25519
- x/oracle/keeper/single.go
  - retrieve consensus key instead of valAddress

----

Closes #XXX

## Summary by CodeRabbit

- **New Features**
  - Introduced advanced signature verification and gas consumption handling for transactions in the Cosmos SDK. This includes new mechanisms for setting public keys, verifying signatures, incrementing sequence numbers, and validating signature counts.

- **Improvements**
  - Enhanced transaction processing with new decorators to better manage signature verification and gas consumption.

- **Compatibility Updates**
  - Updated various package versions to ensure compatibility and improved performance.

### CODE REVIEW

**Comment by leonz789** [comment]
The Lint Code Base error report is for `simulation`, could be ignored

File: app/ante/cosmos/sigverify.go [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+		ak: ak,
+	}
+}
+
+func (spkd SetPubKeyDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	// skip publickkey set for oracle create-price message
+	if isOracleCreatePriceTx(tx) {
+		sigTx, ok := tx.(authsigning.SigVerifiableTx)
+		if !ok {
+			return ctx, sdkerrors.ErrTxDecode.Wrap("invalid tx type")
```
Comment (bwhour) [inline app/ante/cosmos/sigverify.go]: the error is very simple, could be "invalid tx type, expected SigVerifiableTx"

File: app/ante/cosmos/sigverify.go [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+			if !bytes.Equal(signers[i], pk.Address()) {
+				return ctx, sdkerrors.ErrInvalidPubKey.Wrapf("pubKey does not match signer address %s with signer index: %d", signers[i], i)
+			}
+		}
+		return next(ctx, tx, simulate)
+	}
+
+	sigTx, ok := tx.(authsigning.SigVerifiableTx)
+	if !ok {
+		return ctx, sdkerrors.ErrTxDecode.Wrap("invalid tx type")
```
Comment (bwhour) [inline app/ante/cosmos/sigverify.go]: the error is very simple, could be "invalid tx type, expected SigVerifiableTx"

File: x/oracle/keeper/msg_server_create_price_test.go [hunk 3]
```diff
@@ -4,6 +4,8 @@ import (
 	reflect "reflect"
 
 	math "cosmossdk.io/math"
+	dogfoodkeeper "github.com/ExocoreNetwork/exocore/x/dogfood/keeper"
```
Comment (bwhour) [inline x/oracle/keeper/msg_server_create_price_test.go:7]: this keeper has two names : some place is stakingkeeper, some place is dogfoodkeeper

**Review by MaxMustermann2 — APPROVED** [review]

**Review by TimmyExogenous — APPROVED** [review]

File: x/oracle/keeper/common/expected_keepers.go [hunk 4]
```diff
@@ -3,6 +3,7 @@ package common
 import (
 	"cosmossdk.io/math"
 	stakingkeeper "github.com/ExocoreNetwork/exocore/x/dogfood/keeper"
```
Comment (bwhour) [inline x/oracle/keeper/common/expected_keepers.go]: can we unfied stakingkeeper  with dogfoodkeeper ?

File: testutil/keeper/oracle.go [hunk 5]
```diff
@@ -14,7 +14,9 @@ import (
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	typesparams "github.com/cosmos/cosmos-sdk/x/params/types"
-	stakingKeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+
+	// stakingKeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingKeeper "github.com/ExocoreNetwork/exocore/x/dogfood/keeper"
```
Comment (bwhour) [inline testutil/keeper/oracle.go]: can we unfied stakingkeeper with dogfoodkeeper ? and pls remove the comment line stakingKeeper.

Comment (leonz789) [reply testutil/keeper/oracle.go]: updated

File: x/oracle/keeper/common/expected_keepers.go (hunk 4 above)
Comment (leonz789) [reply x/oracle/keeper/common/expected_keepers.go]: updated

File: x/oracle/keeper/msg_server_create_price_test.go (hunk 3 above)
Comment (leonz789) [reply x/oracle/keeper/msg_server_create_price_test.go:7]: updated

File: app/ante/cosmos/sigverify.go (hunk 2 above)
Comment (leonz789) [reply app/ante/cosmos/sigverify.go]: updated

File: app/ante/cosmos/sigverify.go (hunk 1 above)
Comment (leonz789) [reply app/ante/cosmos/sigverify.go]: updated

File: testutil/keeper/oracle.go (hunk 5 above)
Comment (bwhour) [reply testutil/keeper/oracle.go]: this place **stakingKeeper**  did not change to dogfoodKeeper.

Comment (leonz789) [reply testutil/keeper/oracle.go]: updated, thanks !

**Review by bwhour — APPROVED** [review]

**Comment by cloud8little** [comment]
@leonz789 could you fix the lint error?

**Comment by cloud8little** [comment]
Test passed.

**Pull request merged**